modules/base/include/modules/base/processors/inputselector.h

```diff
@@ -113,9 +114,12 @@ InputSelector<Inport, Outport>::InputSelector()
 
 template <typename Inport, typename Outport>
 void InputSelector<Inport, Outport>::updateOptions() {
+    std::string selectedID;
+    if(selectedPort_.size() != 0){
+        selectedID = selectedPort_.getSelectedIdentifier();
+    }
```

Comment:
`TemplateOptionProperty::replaceOptions` should be used. It takes care of maintaining the selected item, if it continues to exist.